src/rt/lifetime.d

```diff
@@ -1193,24 +1186,53 @@ extern (C) void _d_delarray_t(void[]* p, const TypeInfo_Struct ti)
 {
     if (p)
     {
-        assert(!(*p).length || (*p).ptr);
-        if ((*p).ptr)
+        auto bic = __getBlkInfo(p.ptr);
+        auto info = bic ? *bic : GC.query(p.ptr);
+
+        if (info.base && (info.attr & BlkAttr.APPENDABLE))
```

Comment:
Hm... that doesn't seem like the right correlation.

For instance:

```D
auto foo = new Foo[2];
auto bars = foo[0].bars[];
delete bars;
```

I wonder if the type info to call destuctors shouldn't be read from the GC instead of relying on the static type. I didn't review that system that adds finalizer capability to structs. Is this what it is supposed to do?